modules/tensor_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -220,14 +220,35 @@ SingleVariableReturnMappingSolution::internalSolve(const Real effective_trial_st
         {
           if (scalar_upper_bound != max_permissible_scalar && scalar_lower_bound != 0.0)
           {
-            Real frac = 0.5;
-            scalar_increment =
-                (1.0 - frac) * scalar_lower_bound + frac * scalar_upper_bound - scalar_old;
-            modified_increment = true;
-            if (iter_output)
-              *iter_output << "  Trial scalar_increment exceeded bounds.  Setting between "
-                              "lower/upper bounds. frac: "
-                           << frac << std::endl;
+            const Real tol = 1e-14;
+            // Check to avoid divide by zero or overflow in calculation of scalar_increment
+            if (std::abs(1.0 - residual_old / residual) > tol)
+            {
```

Comment:
This doesn't seem quite right. `modified_increment` should be true if you set `scalar_increment`.  Also, I'd replace the check above with absoluteFuzzyEqual(residual_old, residual).

Reply:
Yeah, it's not right.  The fact that this change made one of the Bison example problems run through is probably a fluke.  I have a change that I'll push shortly that seems more correct, but in order to get that Bison example problem to run, I have to accept convergence based on either a small window for the scalar or on a residual below the tolerance.